frontend/src/views/Home.vue

```diff
@@ -16,42 +16,35 @@
 </template>
 
 <script setup lang="ts">
-import { onMounted, provide, ref, shallowReadonly, ShallowRef, shallowRef, watch } from 'vue';
+import { storeToRefs } from 'pinia';
+import { onMounted, ref, watch } from 'vue';
 
 import RequestDialog from '@/components/RequestDialog.vue';
-import useChainCheck from '@/composables/useChainCheck';
-import { createMetaMaskProvider, EthereumProvider } from '@/services/web3-provider';
-import { EthereumProviderKey } from '@/symbols';
+import { createMetaMaskProvider } from '@/services/web3-provider';
+import { useConfiguration } from '@/stores/configuration';
+import { useEthereumProvider } from '@/stores/ethereum-provider';
 
 const criticalErrorMessage = ref('');
-const ethereumProvider = shallowRef<EthereumProvider | undefined>(undefined);
-const readonlyEthereumProvider = shallowReadonly(ethereumProvider);
+const configuration = useConfiguration();
+const ethereumProvider = useEthereumProvider();
 const requestDialogReloadKey = ref(0);
 
-const chainChangeHandler = async () => {
-  const { connectedChainSupported } = useChainCheck(
-    readonlyEthereumProvider as ShallowRef<Readonly<EthereumProvider>>,
-  );
-  const isSupportedChain = await connectedChainSupported();
-  if (!isSupportedChain) {
-    criticalErrorMessage.value = `Connected chain not supported!`;
-  } else {
-    criticalErrorMessage.value = '';
-  }
+const chainChangeHandler = () => {
+  const isSupported = ethereumProvider.chainId in configuration.chains;
+  criticalErrorMessage.value = isSupported ? '' : 'Connected chain not supported!';
 };
 
-provide(EthereumProviderKey, readonlyEthereumProvider);
-
 const resetRequestDialog = () => {
   requestDialogReloadKey.value += 1;
 };
 
 onMounted(async () => {
-  ethereumProvider.value = await createMetaMaskProvider();
-  if (ethereumProvider.value) {
-    watch(ethereumProvider.value.chainId, () => {
-      chainChangeHandler();
-    });
+  ethereumProvider.provider = await createMetaMaskProvider();
+
+  if (ethereumProvider.provider) {
+    const { chainId } = storeToRefs(ethereumProvider);
+
+    watch(chainId, chainChangeHandler);
```

Comment:
Good point. This will be removed anyway with the upcoming WalletConnect integration, but I just fixed it properly. In fact we can always watch for the `chainId`. Nothing special about it. Tested it and works.

Reply:
✔️